### PR TITLE
created contest controllers and set up routes to test functions

### PR DESCRIPTION
**Pull request by brian-glovebox**
### What this PR does (required):
- Create Contest Controller
- Find One Contest Controller
- Find All Contest Controller
- Update Contest Controller 

### Screenshots / Videos (required):
Screen Shot 2021-07-13 at 2 49 54 PM

Screen Shot 2021-07-13 at 2 51 05 PM

### Any information needed to test this feature (required):
- run npm run dev in server folder
- test routes using Postman or Insomnia 

### Any issues with the current functionality (optional):
- Can only update 1 parameter of the contest at a time, maybe change the way the update is ran? Everything currently works.

### CODE REVIEW

File: server/controllers/contest.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+const Contest = require("../models/Contest");
+const asyncHandler = require("express-async-handler");
+
+exports.createContest = asyncHandler(async (req, res) => {
+    try {
+        // expecting req.body to come back in this format.
```
Comment (rajivtitus) [inline server/controllers/contest.js]: Please include the `protect` middleware to your routes, to protect them and access the user's id using the request object, `req.user.id`

File: server/controllers/contest.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        // {
+        // title: req.body.title,
+        // description: req.body.description,
+        // prizeAmount: req.body.prizeAmount,
+        // deadlineDate: req.body.deadlineDate,
+        // not positive how the userid is going to come from the front-end, leaving req.body for now, will change based on how id is being sent from front to back. 
+        // userId: req.body.userId
+        // }
+        const contest = await Contest.create(req.body);
+        res.status(200).json(contest);
```
Comment (rajivtitus) [inline server/controllers/contest.js]: Please update status code to 201 here - for successful resource creation and update it to 500 in the catch block - for a server error :)

Please see for reference: https://www.restapitutorial.com/httpstatuscodes.html

File: server/controllers/contest.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            $set: {
+                [req.body.key]: req.body.update
+            }
+        }, {
+            new: true
+        });
+
+        res.status(200).json(update)
+    } catch (err) {
+        res.status(404).json(err);
```
Comment (rajivtitus) [inline server/controllers/contest.js]: You can return a 404 if we are unable to find a contest with the given id (within the try block). However, the catch block here will return a server error, please update status code to 500 :)
Ditto throughout!

File: server/controllers/contest.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    }
+})
+
+exports.updateContest = asyncHandler(async (req, res) => {
+    try {
+        const update = await Contest.findOneAndUpdate({
+            _id: req.params.id
+        }, {
+            $set: {
```
Comment (rajivtitus) [inline server/controllers/contest.js]: Maybe use request body (`req.body`) here to update multiple fields?

**Review by rajivtitus — COMMENTED** [review]
Looks good for a start!
I've made a few comments, please have a look. Thank you!

File: server/controllers/contest.js (hunk 1 above)
Comment (brian-glovebox) [reply server/controllers/contest.js]: perfect, should have seen that. Will update shortly

File: server/controllers/contest.js (hunk 2 above)
Comment (brian-glovebox) [reply server/controllers/contest.js]: absolute brain fart.

File: server/controllers/contest.js (hunk 4 above)
Comment (brian-glovebox) [reply server/controllers/contest.js]: We were discussing only allowing a contest creator to update one field at a time, and eventually have them to only be able to update certain things, going to make the change and discuss with the whole team if they would eventually like to implement something different. Thank you for all insights.

**Pull request merged**